helpers/response/query_response.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from graphql import GraphQLError
+from utilities.validations import validate_date_range
+
+
+def filter_rooms_by_responses(
+    Query, info, upper_limit_count, lower_limit_count
+):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for response in all_responses:
+        reponse_count = response.total_responses
+        if lower_limit_count <= reponse_count <= upper_limit_count:
+            filtered_responses.append(response)
+    return filtered_responses
+
+
+def filter_response_by_date(Query, info, upper_date_limit, lower_date_limit):
+    all_responses = Query().get_all_reponses(info)
+    filtered_responses = []
+    for responses in all_responses:
+        for response in responses.response:
+            validate_response_dates(
+                response,
+                upper_date_limit, lower_date_limit, filtered_responses
+            )
+        setattr(responses, 'response', filtered_responses)
```

Comment:
What does this line of code do here?

Reply:
The responses is a list. For each response in responses, after validating the date range, this line sets the attribute `responses` of the object responses and adds its value to the filtered_responses.